index.js

```diff
@@ -0,0 +1,365 @@
+//. # sanctuary-argv
+//.
+//. The JavaScript ecosystem has many libraries for parsing command-line
+//. arguments; sanctuary-argv differs from these in a few important ways.
+//.
+//. ## Philosophy
+//.
+//. sanctuary-argv is completely pure. It does not:
+//.
+//.   - reference global variables (such as `process.argv`);
+//.
+//.   - read from standard input;
+//.
+//.   - write to standard output or standard error;
+//.
+//.   - mutate its arguments; or
+//.
+//.   - throw exceptions.
+//.
+//. ## Terminology
+//.
+//.     cut -s -d : -f 1 -- file1 file2 file3 = command
+//.         -s -d : -f 1 -- file1 file2 file3 = arguments
+//.         -s                                = flags = flag names
+//.            -d : -f 1                      = options
+//.            -d   -f                        = option names
+//.               :    1                      = option values
+//.                      --                   = separator
+//.                         file1 file2 file3 = positional arguments
```

Comment:
It took me a while to understand what's going on here. What am I looking at? What the significance of the `cut` command in relation to terminology? How do I read the diagram being painted here? I finally got that you're dissecting an invocation of the `cut` command for the purposes of showing how you'd refer to each segment, with each line kind of being an entry in the dictionary but with the definition on the left.

Maybe a short explanation prior to this dictionary might help readers.

Reply:
I updated this section. Please have another look. :)